pkg/orchestration/controller.go

```diff
@@ -85,65 +85,105 @@ func (c *Controller) Process(ctx *ctrl.ProcessContext) (retriable bool, err erro
 		return false, nil
 	}
 
-	conflict, retriable, bundle, err := c.process(ctx.Logger, state)
+	bundle, external, conflict, retriable, err := c.process(ctx.Logger, state)
 	if conflict || bundle == nil && err == nil {
 		return false, nil
 	}
 
-	conflict, retriable, err = c.handleProcessResult(ctx.Logger, state, bundle, retriable, err)
+	conflict, processRetriable, processErr := c.handleProcessResult(ctx.Logger, state, bundle, retriable, err)
 	if conflict {
 		return false, nil
 	}
+	if processErr != nil {
+		if err != nil && !external {
+			ctx.Logger.Info("Failed to set State status", zap.Error(processErr))
+			return processRetriable || retriable, err
+		} else if external {
```

Comment:
`err != nil && external` ? i.e. `external` has a valid value only if `err` is set.

Reply:
```go
if err != nil {
    if external {
    } else {
    }
}
```
reads better?